scimma_admin/hopskotch_auth/views.py

```diff
@@ -74,5 +97,684 @@ def download(request):
+@login_required
+def remove_user(request):
+    logger.info(f"User {request.user.username} ({request.user.email}) requested to remove user ID " \
+                +request.POST.get("user_id","<unset>")+" from group ID " \
+                +request.POST.get("group_id","<unset>")+" from "+request.META["REMOTE_ADDR"])
+    
+    if not "group_id" in request.POST or len(request.POST["group_id"])==0:
+        return redirect_with_error(request, "Remove a user from a group", 
+                                   "Missing or invalid group ID", "index")
+    if not "user_id" in request.POST or len(request.POST["user_id"])==0:
+        return redirect_with_error(request, "Remove a user from a group", 
+                                   "Missing or invalid user ID", "index")
+    
+    group_id = request.POST["group_id"]
+    user_id=request.POST["user_id"]
+    
+    # only group owners and staff can remove users from groups
+    if not is_group_owner(request.user,group_id) and not request.user.is_staff:
+        return redirect_with_error(request, 
+                                   f"Remove a user from the group with ID {group_id}", 
+                                   "Requester is not a group owner or staff member", 
+                                   "index")
+    
+    try:
+        group = Group.objects.get(id=group_id)
+    except ObjectDoesNotExist as dne:
+        return redirect_with_error(request, 
+                                   f"Remove a user from the group with ID {group_id}", 
+                                   "No such group exists", "index")
+
+    try:
+        target_user = get_user_model().objects.get(id=user_id)
+    except ObjectDoesNotExist as dne:
+        return redirect_with_error(request, 
+                                   f"Remove the user with id {user_id} from the group with ID {group_id}", 
+                                   "No such user exists", "index")
+
+    try:
+        with transaction.atomic():
+            membership = GroupMembership.objects.get(user_id=user_id, group_id=group_id)
+            remove_user_group_permissions(user_id, group_id)
+            membership.delete()
+    except ObjectDoesNotExist as dne:
+        # apparently we need do nothing
+        pass
+
+    logger.info(f"Removed user {target_user.username} ({target_user.email}) from" \
+                +f" group {group.name}")
+    messages.info(request, "Removed user from group.")
+    base_edit_url = reverse("edit_group")
+    return redirect(base_edit_url+"?group_id="+group_id)
+
+
+@require_POST
+@login_required
+def create_topic(request):
+    logger.info(f"User {request.user.username} ({request.user.email}) requested to create a topic with name " \
+                +request.POST.get("topic_name","<unset>")+" owned by group ID " \
+                +request.POST.get("group_id","<unset>")+" from "+request.META["REMOTE_ADDR"])
+    
+    if not "group_id" in request.POST or not validate_topic_name(request.POST["group_id"]):
```

Comment:
`validate_topic_name` doesn't look right here. Validate the group ID, I think.

Reply:
Indeed, this makes no sense and was probably a copy-and-paste mistake. As noted in the comment a few lines down, `is_group_owner` implicitly validates the `group_id`, so I don't think this is needed at all.